From b3cb6a08c82deacb905459b66f16eda35f0c3779 Mon Sep 17 00:00:00 2001
From: Michael Palimaka <kensington@gentoo.org>
Date: Wed, 9 Nov 2016 21:42:08 +1100
Subject: [PATCH] Fix build with ENABLE_NETWORK_CLIPBOARD_SHARING enabled.

qlippernetwork.cpp expects network-related variables to be available if
ENABLE_NETWORK_CLIPBOARD_SHARING is defined, but qlippernetwork.h defined those
variables if ENABLE_NETWORK_CLIPBOARD_SHARING was *un*defined.

This also adds a missing include that is no longer available transitively via
QtNetwork/QUdpSocket.
---
 src/qlippernetwork.h | 4 ++--
 src/qlippertypes.h   | 2 +-
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/qlippernetwork.h b/src/qlippernetwork.h
index 9163cf8..ce1ac9a 100644
--- a/src/qlippernetwork.h
+++ b/src/qlippernetwork.h
@@ -21,7 +21,7 @@ with this program; if not, write to the Free Software Foundation, Inc.,
 #define QLIPPERNETWORK_H
 
 #include <QtCore/QObject>
-#ifndef ENABLE_NETWORK_CLIPBOARD_SHARING
+#ifdef ENABLE_NETWORK_CLIPBOARD_SHARING
 #include <QtNetwork/QUdpSocket>
 #endif
 #include "qlippertypes.h"
@@ -38,7 +38,7 @@ public slots:
     void sendData(const ClipboardContent &value);
 
 private:
-#ifndef ENABLE_NETWORK_CLIPBOARD_SHARING
+#ifdef ENABLE_NETWORK_CLIPBOARD_SHARING
     QUdpSocket *m_socket;
     QString m_id;
 #endif
diff --git a/src/qlippertypes.h b/src/qlippertypes.h
index b55f0e1..c97c311 100644
--- a/src/qlippertypes.h
+++ b/src/qlippertypes.h
@@ -2,7 +2,7 @@
 #define QLIPPERTYPES_H
 
 #include <QMetaType>
-
+#include <QHashIterator>
 
 typedef QHash<QString,QByteArray> ClipboardContent;
 typedef QHashIterator<QString,QByteArray> ClipboardContentIterator;
-- 
2.7.3

